src/main/java/commands/Command.java

```diff
@@ -0,0 +1,52 @@
+package commands;
+
+import java.util.ArrayList;
+
+import storage.Storage;
+import tasks.TaskList;
+import ui.Ui;
+
+/**
+ * The Command class represents the action to be done.
+ */
+public abstract class Command {
+
+    private final ArrayList<String> input;
+
+    /**
+     * Constructs the Command object.
+     *
+     * @param s the entire line of user input
+     */
+    public Command(ArrayList<String> s) {
+        this.input = s;
```

Comment:
Should variables with single-character names be used for nested loops only?

Reply:
changed to "userInput", thanks for pointing it out!